### PR TITLE
Add flatpak Discord Rich Presence support

### PR DESCRIPTION
**Pull request by Parnassius**

### CODE REVIEW

**Comment by cpba** [comment]
Thank you for taking the time to do this.

Could you elaborate a bit on what this feature does and how to test it? 

The discord website describes it as a source of viral advertising for games, is it intended to show other people on discord that you are playing games on dolphin? Does it do anything else?

**Comment by Parnassius** [comment]
Yes, it allows to show other people what game you are playing on dolphin. Upstream dolphin already has this feature, this pull request only adds support for discord installed as a flatpak. See https://github.com/flathub/com.discordapp.Discord/issues/29 and https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc) for more informations.

I tested both a local build and the build provided by the bot, and it appears to be working. To test it yourself you should install discord from flathub, create an account if you don't already have one, and then run dolphin. 

If you just created a new account probably the easiest way to test is to create a new server (green + on the left side of the window, you only have to provide a name).
On your profile (on the right side) should then appear something like this:

screenshot NRR9R0hNJHrBdrwf

screenshot fuLXWH8kDi4GZUDr

**Comment by TingPing** [comment]
Desktop file needs updated.

File: org.DolphinEmu.dolphin-emu.json [hunk 1]
```diff
@@ -83,6 +85,10 @@
         {
           "type": "file",
           "path": "appdata.xml"
+        },
+        {
+          "type": "file",
```
Comment (TingPing) [inline org.DolphinEmu.dolphin-emu.json]: You can just use `type: script`.

**Comment by LeandroStanger** [comment]
bot, build org.DolphinEmu.dolphin-emu

File: org.DolphinEmu.dolphin-emu.json [hunk 2]
```diff
@@ -68,9 +68,10 @@
         "/share/man"
       ],
       "post-install": [
-        "install -D dolphin-emu.sh /app/bin/dolphin-emu.sh",
+        "install -D dolphin-emu-wrapper /app/bin/dolphin-emu-wrapper",
         "install -Dm644 appdata.xml /app/share/appdata/dolphin-emu.appdata.xml",
-        "sed -i -e 's/\"2048\"/\"512\"/g' /app/share/icons/hicolor/scalable/apps/dolphin-emu.svg"
+        "sed -i -e 's/\"2048\"/\"512\"/g' /app/share/icons/hicolor/scalable/apps/dolphin-emu.svg",
+        "sed -i -e 's|Exec=dolphin-emu|Exec=/app/bin/dolphin-emu-wrapper|' /app/share/applications/dolphin-emu.desktop"
```
Comment (TingPing) [inline org.DolphinEmu.dolphin-emu.json]: `desktop-file-edit` is a bit cleaner than `sed`.

**Comment by cpba** [comment]
@Parnassius your builds are failing because of the new flathub appdata requirements, I've been meaning to fix it for a while but I haven't gotten around to doing it yet.

**Comment by cpba** [comment]
bot, build

**Comment by TingPing** [comment]
Looks fine. @cpba can fix the appdata then we'll rebase this.

**Comment by cpba** [comment]
I'm pretty sure I fixed it yesterday before bed, you can rebase.

**Comment by gasinvein** [comment]
Before this is merged, I hope to change discord-rpc socket path from `xdg-run/discord/ipc-0` to `xdg-run/app/com.discordapp.Discord/discord-ipc-0`, see flathub/com.discordapp.Discord#68.

**Comment by cpba** [comment]
I understand that this pull request is waiting on https://github.com/flathub/com.discordapp.Discord/pull/68 before it can be merged?

**Comment by gasinvein** [comment]
The change required after that PR would be trivial, so it's probably not a blocker. Just be aware of its state.

**Comment by cpba** [comment]
My concern is that if I merge this now, anybody who starts using it will
see it break when they update discord.

Would this be the case?

**Comment by gasinvein** [comment]
Yes.

**Comment by TingPing** [comment]
Merged the Discord change so we can update this.

**Comment by TingPing** [comment]
This has a merge conflict and needs to be rebased (feel free to squash).

File: org.DolphinEmu.dolphin-emu.json [hunk 3]
```diff
@@ -83,6 +86,16 @@
         {
           "type": "file",
           "path": "appdata.xml"
+        },
+        {
+          "type": "script",
+          "commands": [
+            "for i in {0..9}; do",
```
Comment (TingPing) [inline org.DolphinEmu.dolphin-emu.json]: Only `0` is ever used AFAIK?

Comment (gasinvein) [reply org.DolphinEmu.dolphin-emu.json]: The docs says that
>Discord can be on any pipe ranging from discord-ipc-0 to discord-ipc-9

but we a exporting only `0` anyway.

Comment (Parnassius) [reply org.DolphinEmu.dolphin-emu.json]: I copied that snippet from https://github.com/flathub/com.discordapp.Discord/wiki/Rich-Precense-(discord-rpc), should I create only ``discord-ipc-0``?

Comment (gasinvein) [reply org.DolphinEmu.dolphin-emu.json]: IMO better leave it as is, redundant symlinks inside the sandbox are harmless.

**Pull request merged**